api/design/resource.go

```diff
@@ -38,21 +96,212 @@ var _ = Service("resource", func() {
+				Example("version", "0.1")
+			})
+
+			Required("catalog", "kind", "name", "version")
+		})
+		Result(func() {
+			Attribute("location", String, "Redirect URL", func() {
+				Format(FormatURI)
+			})
+			Required("location")
+		})
+
+		HTTP(func() {
+			GET("/resource/{catalog}/{kind}/{name}/{version}/readme")
+			Response(StatusFound, func() {
+				Header("location")
+			})
 
-			Response(StatusOK)
-			Response("internal-error", StatusInternalServerError)
-			Response("not-found", StatusNotFound)
+		})
+	})
+
+	Method("ByCatalogKindNameVersionYaml", func() {
+		Description("Find resource README using name of catalog & name, kind and version of resource")
+		Payload(func() {
+			Attribute("catalog", String, "name of catalog", func() {
+				Example("catalog", "tektoncd")
+			})
+			Attribute("kind", String, "kind of resource", func() {
+				Enum("task", "pipeline")
+			})
+			Attribute("name", String, "name of resource", func() {
+				Example("name", "buildah")
+			})
+			Attribute("version", String, "version of resource", func() {
+				Example("version", "0.1")
+			})
+
+			Required("catalog", "kind", "name", "version")
+		})
+		Result(func() {
+			Attribute("location", String, "Redirect URL", func() {
+				Format(FormatURI)
+			})
+			Required("location")
+		})
+
+		HTTP(func() {
+			GET("/resource/{catalog}/{kind}/{name}/{version}/yaml")
+			Response(StatusFound, func() {
+				Header("location")
+			})
+
+		})
+	})
+
+	Method("ByVersionId", func() {
+		Description("Find a resource using its version's id")
+		Payload(func() {
+			Attribute("versionID", UInt, "Version ID of a resource's version", func() {
+				Example("versionID", 1)
+			})
+			Required("versionID")
+		})
+		Result(func() {
+			Attribute("location", String, "Redirect URL", func() {
+				Format(FormatURI)
+			})
+			Required("location")
+		})
+
+		HTTP(func() {
+			GET("/resource/version/{versionID}")
+			Response(StatusFound, func() {
+				Header("location")
+			})
+		})
+	})
+
+	Method("ByCatalogKindName", func() {
+		Description("Find resources using name of catalog, resource name and kind of resource")
+		Payload(func() {
+			Attribute("catalog", String, "name of catalog", func() {
+				Example("catalog", "tektoncd")
+			})
+			Attribute("kind", String, "kind of resource", func() {
+				Enum("task", "pipeline")
+			})
+			Attribute("name", String, "Name of resource", func() {
+				Example("name", "buildah")
+			})
+			Attribute("pipelinesversion", String, "To find resource compatible with a Tekton pipelines version, use this param", func() {
+				Pattern(types.PipelinesVersionRegex)
+				Example("pipelinesversion", "0.21.0")
+			})
+			Required("catalog", "kind", "name")
+		})
+		Result(func() {
+			Attribute("location", String, "Redirect URL", func() {
+				Format(FormatURI)
+			})
+			Required("location")
+		})
+
+		HTTP(func() {
+			GET("/resource/{catalog}/{kind}/{name}")
+
+			Param("pipelinesversion")
+
+			Response(StatusFound, func() {
```

Comment:
just trying to understand
so we add the url we want to redirect in response header as `Location`
and goa will redirect to that API?

Reply:
yes so we are adding an attribute in `Result` as being the value for the location header and then we map that to the Location response header to which it redirects to the value which we set in the service logic